src/ast.scm

```diff
@@ -523,6 +523,16 @@
   (and (if one (length= e 3) (length> e 2))
        (eq? (car e) 'meta) (memq (cadr e) '(nospecialize specialize))))
 
+(define (method-meta? e)
+  (and (length> e 1)
+       (eq? (car e) 'meta)
+       (any (lambda (x)
+              (memq x '(inline noinline aggressive_constprop no_constprop propagate_inbounds)))
```

Comment:
This will keep any `meta` expr that contains one of these, even if it also contains other things, e.g. it will propagate `(meta inline pure)`. I'm not sure that ever happens in practice, but we might need to handle it.

Reply:
Addressed. I also added a similar support for `:purity` expression (for the effect analysis).